mxfusion/components/distributions/wishart.py

```diff
@@ -214,19 +214,34 @@ def draw_samples(self, degrees_of_freedom, scale, rv_shape, num_samples=1, F=Non
         # Cholesky of L
         L = F.linalg.potrf(scale)
 
-        dof = degrees_of_freedom.asnumpy().ravel()[0]
+        full_shape = (num_samples,) + rv_shape
+
+        import itertools
 
         # Create the lower triangular matrix A
-        A = F.zeros((dof, dof), dtype=scale.dtype)
-        for j in range(dof):
-            A[j, j] = np.sqrt(np.random.chisquare(df=dof - j))
-            for k in range(j + 1, dof):
-                A[k, j] = np.random.normal()
+        A = F.zeros(full_shape, dtype=scale.dtype)
+        for i in range(num_samples):
+            # Need to have the full shape
+            for extra_dims in itertools.product(*map(range, rv_shape[:-2])):
+                for j in range(rv_shape[-2]):
+                    shape = degrees_of_freedom.asnumpy().ravel()[0] - j
+                    try:
+                        ix = (i, ) + extra_dims + (j, j)
+                        A[ix] = F.sqrt(F.sum(F.power(F.random.normal(shape=shape), 2)))
+                    except Exception as e:
```

Comment:
What kind of exception are you expecting here? This is a weird try/catch since you don't do anything with it. If you know the reason for this, can you add a message to this? Better yet, raise an InferenceError (mxfusion.common.errors) with the relevant message.

Reply:
Oops sorry that's debugging code. FYI I do that so I can place a breakpoint on the `raise e` line to drop into the debugger if any exception is triggered - it's a pattern I use often. I've removed this.